types/simulation/types.go

```diff
@@ -95,11 +101,35 @@ func NewOperationMsg(msg sdk.Msg, ok bool, comment string, cdc *codec.ProtoCodec
 		moduleName = msgType
 	}
 
-	if legacyMsg, okType := msg.(legacytx.LegacyMsg); okType {
-		return NewOperationMsgBasic(moduleName, msgType, comment, ok, legacyMsg.GetSignBytes())
+	// Below is logic added for aminojson.Encoder compatibility with gogoproto messages.
+	// x/tx/signing/aminojson cannot marshal sdk.Msg (an alias to gogoproto.message) directly since this type is not
+	// protoreflect enabled like the std library google.golang.org/protobuf/proto.Message is.
+	//
+	// So we just convert to a dynamicpb message and marshal that directly
+	resolver := gogoproto.HybridResolver
+	desc, err := resolver.FindDescriptorByName(protoreflect.FullName(gogoproto.MessageName(msg)))
+	if err != nil {
+		panic(fmt.Errorf("failed to find proto descriptor for %s: %w", msgType, err))
+	}
+
+	dynamicMsgType := dynamicpb.NewMessageType(desc.(protoreflect.MessageDescriptor))
+	dynamicMsg := dynamicMsgType.New().Interface()
+	gogoBytes, err := gogoproto.Marshal(msg)
+	if err != nil {
+		panic(fmt.Errorf("failed to marshal msg: %w", err))
+	}
+	err = proto.Unmarshal(gogoBytes, dynamicMsg)
+	if err != nil {
+		panic(fmt.Errorf("failed to unmarshal msg: %w", err))
+	}
+
+	encoder := aminojson.NewEncoder(aminojson.EncoderOptions{FileResolver: resolver})
+	jsonBytes, err := encoder.Marshal(dynamicMsg)
+	if err != nil {
+		panic(fmt.Errorf("failed to marshal msg: %w", err))
 	}
```

Comment:
Since this will be pretty common let's create some helpers for it. I would even go as far as saying that we can add an `MarshalAminoJSON` method to `codec.Codec` - it already has the proper file resolver from `InterfaceRegistry`

Reply:
Are you suggesting adding `MarshalAminoJSON` to [JSONCodec](https://github.com/cosmos/cosmos-sdk/blob/cb1641c1b5f844190723f34db7f73b85c8d43580/codec/codec.go#L80)?